src/libopensc/pkcs15-pin.c

```diff
@@ -367,6 +367,7 @@ int sc_pkcs15_verify_pin(struct sc_pkcs15_card *p15card,
 	r = sc_pin_cmd(card, &data, &auth_info->tries_left);
 	sc_log(ctx, "PIN cmd result %i", r);
 	if (r == SC_SUCCESS)
+	if (r == SC_SUCCESS && pin_obj->usage_counter == 0)
```

Comment:
As for me,
here PIN is validated with the value that comes "from the outer level" (application), and so it's natural that cache is completely reinitialized, without any regards onto current _usage-counter_.